### PR TITLE
DAOS-12674 test: nvme/enospace.py test not waiting for thread completion

### PR DESCRIPTION
**Pull request by dinghwah**
Description: update script enospace.py to complete the job before exit.

Test-tag: der_enospace
Test-repeat: 30
Doc-only: false
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by phender — CHANGES_REQUESTED** [review]
We should use a threading.Event to trigger when to stop running the ior reads - also ensuring at least one is read is run:
```
        ior_bg_cmd.flags.update(self.ior_read_flags)
        stop_looping = False
        while not stop_looping:
            try:
                job_manager.run()
            except (CommandFailure, TestFail):
                self.test_result.append("FAIL - ior read")
                break
            stop_looping = event.wait(0)
```

File: src/tests/ftest/nvme/enospace.py [hunk 1]
```diff
@@ -206,12 +220,14 @@ def test_enospace_lazy_with_bg(self):
 
         :avocado: tags=all,full_regression
         :avocado: tags=hw,medium,ib2
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/enospace.py:222]: Pre-existing - if you have to push again: `ib2` is no longer used. here and other methods below
```suggestion
        :avocado: tags=hw,medium
```

File: src/tests/ftest/nvme/enospace.py [hunk 2]
```diff
@@ -109,21 +111,24 @@ def ior_bg_thread(self):
...
         ior_bg_cmd.flags.update(self.ior_read_flags)
-        while True:
+        stop_looping = False
+        while not stop_looping:
             try:
                 job_manager.run()
             except (CommandFailure, TestFail):
+                self.test_result.append("FAIL - ior read")
                 break
+        stop_looping = event.wait(1)
```
[the remark this answers is not in the document]
Comment (dinghwah) [reply src/tests/ftest/nvme/enospace.py:131]: Dalton yes.
Thanks.

**Review by shimizukko — APPROVED** [review]

**Review by phender — APPROVED** [review]
It would be nice to include some logging that logs:
- Setting of the event
- Detecting the event
- Indicating how many ior reads are run in the background

**Pull request merged**